src/main/kotlin/eu/theblob42/idea/whichkey/WhichKeyTypeActionHandler.kt

```diff
@@ -49,16 +49,15 @@ class WhichKeyTypeActionHandler(private val vimTypedActionHandler: VimTypedActio
                 if (ideFrame != null) {
                     val startTime = System.currentTimeMillis() // save start time for the popup delay
 
-                    val mappingState = commandState.mappingState
-                    val typedKeySequence = mappingState.keys + listOf(KeyStroke.getKeyStroke(charTyped))
+                    val typedKeySequence = commandState.commandBuilder.keys + listOf(KeyStroke.getKeyStroke(charTyped))
                     val mappingMode = editor.vim.mode.toMappingMode()
                     val nestedMappings = MappingConfig.getNestedMappings(mappingMode, typedKeySequence)
 
                     if (nestedMappings.isEmpty()) {
                         if (mappingMode != MappingMode.INSERT
                             && !MappingConfig.processWithUnknownMapping(mappingMode, typedKeySequence)) {
                             // reset the mapping state, do not open a popup & ignore the next call to `execute`
-                            mappingState.resetMappingSequence()
+                            commandState.mappingState.resetMappingSequence()
```

Comment:
No idea if this is still correct, maybe you can help out @TheBlob42 ?

Reply:
I need to find the time to properly go through all this code that I haven't touched in over a year. I hope I then can give a better answer